pkg/services/pluginsintegration/pluginstore/store.go

```diff
@@ -33,11 +34,36 @@ type Service struct {
 	pluginRegistry registry.Service
 	pluginLoader   loader.Service
 	pluginSources  sources.Registry
+	loadOnStartup  bool
 }
 
 func ProvideService(pluginRegistry registry.Service, pluginSources sources.Registry,
-	pluginLoader loader.Service) *Service {
-	return New(pluginRegistry, pluginLoader, pluginSources)
+	pluginLoader loader.Service, features featuremgmt.FeatureToggles) (*Service, error) {
+	if features.IsEnabledGlobally(featuremgmt.FlagPluginStoreServiceLoading) {
```

Comment:
Should we make the old functionality the default toggle-on state (or just have it default to `true`) so that when the commit rolls out, we don't have to also trigger more deploys with a FT change?